### PR TITLE
Fix an integration test flake on NodeAfffinity ScorePlugin

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind flake
/sig scheduling
/priority critical-urgent

**What this PR does / why we need it**:

The behavior of Score plugins are not deterministic as the Scores of a plugin can be easily diluted by others.

This PR constructs a scheduler profile by disabling all Score plugins except for `NodeAffiniy`, so the final score is only impacted by `NodeAffiniy` Score plugin.

**Which issue(s) this PR fixes**:

Fixes NodeAffinity flake in #93485.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/scheduler/priorities_test.go [hunk 1]
```diff
@@ -36,7 +39,29 @@ import (
...
+	testCtx := testutils.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestMaster(t, "node-affinity", nil),
+		false,
+		nil,
+		0,
+		scheduler.WithProfiles(prof),
+	)
+	testutils.SyncInformerFactory(testCtx)
+	go testCtx.Scheduler.Run(testCtx.Ctx)
```
Comment (ahg-g) [inline test/integration/scheduler/priorities_test.go:64]: I expect all other score tests have the same issue, can we wrap this around a function and use it everywhere?:

```
func initTestWithPlugins(name string, enabledPlugins []schedulerconfig.Plugin)  *TestContext {
        prof := schedulerconfig.KubeSchedulerProfile{
		SchedulerName: v1.DefaultSchedulerName,
		Plugins: &schedulerconfig.Plugins{
			Score: &schedulerconfig.PluginSet{
				Enabled:enabledPlugins,
				Disabled: []schedulerconfig.Plugin{
					{Name: "*"},
				},
			},
		},
	}
	testCtx := testutils.InitTestSchedulerWithOptions(
		t,
		testutils.InitTestMaster(t, name, nil),
		false,
		nil,
		0,
		scheduler.WithProfiles(prof),
	)
	testutils.SyncInformerFactory(testCtx)
	go testCtx.Scheduler.Run(testCtx.Ctx)
        return testCtx
}
```

**Comment by ahg-g** [comment]
> The behavior of Score plugins are not deterministic as the Scores of a plugin can be easily diluted by others. In the flake test, an explicit weight: 20 can only mitigate the flake.

Where was the weight of 20 set? or are you mentioning that as an example?

File: test/integration/scheduler/priorities_test.go (hunk 1 above)
Comment (Huang-Wei) [reply test/integration/scheduler/priorities_test.go:64]: sure, will update.

**Comment by Huang-Wei** [comment]
Ah, I misread the NodeAffinity weight as the plugin weight... Updated the wordings.

**Comment by ahg-g** [comment]
Thanks, please squash!

**Comment by Huang-Wei** [comment]
Verify job timed out.

/retest

**Pull request merged**